cmake/scripts/common/HandleDepends.cmake

```
@@ -231,7 +231,7 @@ function(add_addon_depends addon searchpath)
                                     -DCMAKE_INCLUDE_PATH=${OUTPUT_DIR}/include)
             endif()
 
-            set(DOWNLOAD_DIR ${BUILD_DIR}/download)
+            set(DOWNLOAD_DIR ${BUILD_DIR}/${id}/download)
```

This may conflict with the `externalproject_add` which already uses `${BUILD_DIR}/${id}` folder.
```suggestion
            set(DOWNLOAD_DIR ${BUILD_DIR}/download/${id})
```